src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/ResendStaffNotificationsCoreCaseDataTest.java

```diff
@@ -0,0 +1,296 @@
+                "state", "open",
+                "sortDirection", "desc"))
+            )
+        ).willReturn(listOfCaseDetailsWithDefendant());
+
+        makeRequest(CASE_REFERENCE, "more-time-requested").andExpect(status().isOk());
+
+        verify(notificationClient).sendEmail(eq("staff-more-time-requested-template"),
+            eq("recipient@example.com"),
+            any(),
+            eq("more-time-requested-notification-to-staff-" + CASE_REFERENCE));
+    }
+
+    @Test
+    public void shouldRespond409AndNotProceedForResponseSubmittedEventWhenResponseNotSubmitted() throws Exception {
+        given(coreCaseDataApi.searchForCaseworker(
+            any(),
+            any(),
+            any(),
+            any(),
+            any(),
+            eq(ImmutableMap.of("case.referenceNumber", CASE_REFERENCE,
+                "page", PAGE,
+                "state", "open",
+                "sortDirection", "desc"))
+            )
+        ).willReturn(listOfCaseDetails());
+
+        makeRequest(CASE_REFERENCE, "response-submitted").andExpect(status().isConflict());
+
+        verify(emailService, never()).sendEmail(any(), any());
+    }
+
+    @Test
+    public void shouldRespond200AndSendNotificationsForResponseSubmittedEvent() throws Exception {
+        given(coreCaseDataApi.searchForCaseworker(
+            any(),
+            any(),
+            any(),
+            any(),
+            any(),
+            eq(ImmutableMap.of("case.referenceNumber", CASE_REFERENCE,
+                "page", PAGE,
+                "state", "open",
+                "sortDirection", "desc"))
+            )
+        ).willReturn(listOfCaseDetailsWithDefResponse());
+
+        makeRequest(CASE_REFERENCE, "response-submitted").andExpect(status().isOk());
+
+        verify(emailService).sendEmail(eq("sender@example.com"), emailDataArgument.capture());
+
+        assertThat(emailDataArgument.getValue().getTo()).isEqualTo("recipient@example.com");
+        assertThat(emailDataArgument.getValue().getSubject())
+            .isEqualTo("Civil Money Claim defence submitted: John Rambo v John Smith " + CASE_REFERENCE);
+        assertThat(emailDataArgument.getValue().getMessage()).contains(
+            "The defendant has submitted a full defence which is attached as a PDF",
+            "Email: j.smith@example.com",
+            "Mobile number: 07873727165"
+        );
+    }
+
+    @Test
+    public void shouldRespond200AndSendNotificationsForCCJRequestedEvent() throws Exception {
+        given(coreCaseDataApi.searchForCaseworker(
+            any(),
+            any(),
+            any(),
+            any(),
+            any(),
+            eq(ImmutableMap.of("case.referenceNumber", CASE_REFERENCE,
+                "page", PAGE,
+                "state", "open",
+                "sortDirection", "desc"))
+            )
+        ).willReturn(listOfCaseDetailsWithCCJ());
+
+        makeRequest(CASE_REFERENCE, "ccj-request-submitted").andExpect(status().isOk());
```

Comment:
I was surprised to see there is no check whether CCJ was actually requested before we send CCJ requested event. Not part of this story but would be good to improve in the future.